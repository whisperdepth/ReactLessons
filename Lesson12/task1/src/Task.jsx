import React from 'react';
import classNames from 'classnames';

const Task = ({ done, text}) => {
    return (
      <li className={classNames('list-item', {'list-item_done': done})}>
        <input type='checkbox' className='list-item__checkbox' checked={done} />
        <span className='list-item__text'>{text}</span>
        <button className='list-item__delete-btn' />
      </li>
    );
}

export default Task;
