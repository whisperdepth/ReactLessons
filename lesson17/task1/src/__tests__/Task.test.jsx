import React from "react";
import { shallow } from "enzyme";
import Task from "../Task";

describe("<Task />", () => {
  it("should display done task", () => {
    const props = {
      text: "task1",
      id: "1",
      done: true,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapper = shallow(<Task {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should display undone task", () => {
    const props = {
      text: "task1",
      id: "1",
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapper = shallow(<Task {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should update task on checkbox checked", () => {
    const props = {
      text: "task1",
      id: "1",
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapper = shallow(<Task {...props} />);
    wrapper.find(".list-item__checkbox").simulate("change");

    expect(props.onChange).toBeCalledWith("1");
  });

  it("should delete task on delete btn click", () => {
    const props = {
        text: "task1",
        id: "1",
        done: false,
        onChange: jest.fn(),
        onDelete: jest.fn(),
      };
      const wrapper = shallow(<Task {...props} />);
      wrapper.find(".list-item__delete-btn").simulate("click");

    expect(props.onDelete).toBeCalledWith("1");
  });
});
