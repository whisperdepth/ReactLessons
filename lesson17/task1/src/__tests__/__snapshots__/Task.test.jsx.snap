// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> should display done task 1`] = `
<li
  className="list-item list-item_done"
>
  <input
    checked={true}
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    task1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<Task /> should display undone task 1`] = `
<li
  className="list-item null"
>
  <input
    checked={false}
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    task1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;
